packages/zero-client/src/client/query-manager.ts

```diff
@@ -358,12 +369,39 @@ export class QueryManager {
     if (entry.count === 0) {
       this.#recentQueries.add(astHash);
       if (this.#recentQueries.size > this.#recentQueriesMaxSize) {
-        const lruAstHash = this.#recentQueries.values().next().value;
-        assert(lruAstHash);
-        this.#queries.delete(lruAstHash);
-        this.#recentQueries.delete(lruAstHash);
-        this.#queueQueryChange({op: 'del', hash: lruAstHash});
+        const lruQueryID = this.#recentQueries.values().next().value;
+        assert(lruQueryID);
+        this.#queries.delete(lruQueryID);
+        this.#recentQueries.delete(lruQueryID);
+        this.#queryMetrics.delete(lruQueryID);
+        this.#queueQueryChange({op: 'del', hash: lruQueryID});
       }
     }
   }
+
+  get metrics() {
```

Comment:
a comment on this method about what metrics to expect to be returned would be useful